# Homework 1
### Rolando Santos
### https://github.com/rsantos2032/DSCI401/
### 2023-08-31

In [40]:
# Loading the Dataset
import pandas as pd
cubs = pd.read_csv("https://raw.githubusercontent.com/gjm112/DSCI401/main/data/cubs_all_time.csv")
cubs = cubs.rename(columns = {"G":"Games", "W":"Wins", "L":"Losses", "R":"Runs"}) # Renaming column names for clarity

### Question 1: How many total games have the Cubs won and lost between 1876 and 2022?

In [41]:
# Filtering our dataset to only include rows with years between 1876 and 2022
cubs_filtered_years = cubs.loc[(cubs.Year > 1875) & (cubs.Year < 2023), ["Year", "Games", "Wins", "Losses"]]
cubs_filtered_years["Wins"].sum()

11141

After summing the entire wins column, we can see that the total number of cubs wins in the dataset is 11141.

### Question 2: What year did the Cubs score the most runs? What year did the Cubs score the fewest runs? Do you have any thoughts about the year that the Cubs scored the fewest runs?

In [42]:
# Selecting Data with might find relevant
cubs_sort = cubs[["Year", "Games", "Wins", "Losses", "Runs"]]
cubs_sort.sort_values(by = "Runs", ascending = False)[0:5] 

Year  Games  Wins  Losses  Runs
128  1894    137    57      75  1056
92   1930    156    90      64   998
93   1929    156    98      54   982
136  1886    126    90      34   900
133  1889    136    67      65   867

The cubs scored their most runs in the year 1894 with a total of 1056 runs.

In [43]:
cubs_sort.sort_values(by = "Runs", ascending = True)[0:5] 

Year  Games  Wins  Losses  Runs
2    2020     60    34      26   265
145  1877     60    26      33   366
41   1981    106    38      65   370
144  1878     61    30      30   371
143  1879     83    46      33   437

The cubs scored their least amount of runs in the year 2020 with a total of 265. A reason for this could be due to the low amount of games played that year, 60, compared to the years they scored a lot of runs, which had more than double the amount of games played.

### Question 3: In how many seasons was the Cubs total attendance (i.e. the variable Attendance) over 3 million?

In [44]:
cubs_filtered_attendance = cubs.loc[cubs.Attendance > 3000000, ["Year", "Games", "Wins", "Losses", "Attendance"]]
cubs_filtered_attendance.reset_index(drop = True) # Done solely to clearly show the number of rows

Year  Games  Wins  Losses  Attendance
0   2019    162    84      78   3094865.0
1   2018    163    95      68   3181089.0
2   2017    162    92      70   3199562.0
3   2016    162   103      58   3232420.0
4   2011    162    71      91   3017966.0
5   2010    162    75      87   3062973.0
6   2009    161    83      78   3168859.0
7   2008    161    97      64   3300200.0
8   2007    162    85      77   3252462.0
9   2006    162    66      96   3123215.0
10  2005    162    79      83   3099992.0
11  2004    162    89      73   3170154.0

In [45]:
cubs_filtered_attendance.shape[0]

12

After filtering the data set we end up with a data frame with 12 rows, also confirmed with shape()[0] returning 12, thus we can conclude that there were 12 years/seasons where the cubs had an attendance of over 3 million.